cub/cub/device/device_reduce.cuh

```diff
@@ -195,6 +196,8 @@ struct DeviceReduce
     T init,
     cudaStream_t stream = 0)
   {
+    detail::NVTXGuard _(__func__);
```

Comment:
Important: I am personally not a fan of `_` as an identifier.

Also I would like this to be within a macro, so we can hide `__func__` and potentially move towards `source_location` at a later date.
```suggestion
    CUB_NVTX_GUARD;
```
We should also add a void cast to the macro, as some compiler will complain about the unused variable

Reply:
I can name it anything else. I picked `_` because of its semantics to be a placeholder without name (cf. [p2169](https://p2169)).

I anticipated `source_location`, which is why I added a dedicated ctor. For now I don't know yet what C++ standard I can expect for this feature.

> We should also add a void cast to the macro, as some compiler will complain about the unused variable

Which is exactly the samantics of `_` in C++23. And why a `(void)` and not a `[[maybe_unused]]`?

Reply:
Yes, but this PR also needs inline variables for global setup and teardown of the `CUB` NVTX domain. Are there other setup and teardown functions I could use instead?